pkg/credentialprovider/apis/dockerauth/v1alpha1/types.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type DockerAuthPluginConfig struct {
+	metav1.TypeMeta `json:",inline"`
+
+	Plugins []DockerAuthPlugin `json:"plugins"`
+}
+
+// DockerAuthPlugin is used by the kubelet container runtime to match the
+// image property string (from the container spec) with exec-based credential provider
+// plugins that provide container registry credentials.
+type DockerAuthPlugin struct {
+	// name is the name of the credential plugin.  It must match the name of the
+	// executable as seen by the kubelet.
+	Name string `json:"name"`
```

Comment:
Is this the full file path? eg. "/foo/bar/a really long file path I don't want to type/authSideCar"

Reply:
I was thinking it's just the name of the executable and it's expected that it's in the kubelet's PATH or we pass a bin dir to kubelet similar to CNI.